driver/driver.go

```diff
@@ -202,12 +231,12 @@ func (d *ContrailDriver) CreateEndpoint(req *network.CreateEndpointRequest) (*ne
 		fmt.Printf("%v: %v\n", k, v)
 	}
 
-	tenant, netName, err := d.tenantAndNetnameFromDockerNetwork(req.NetworkID)
+	meta, err := d.networkMetaFromDockerNetwork(req.NetworkID)
 	if err != nil {
 		return nil, err
 	}
 
-	contrailNetwork, err := d.controller.GetNetwork(tenant, netName)
+	contrailNetwork, err := d.controller.GetNetwork(meta.tenant, meta.network)
```

Comment:
Maybe you could rewrite GetNetwork to handle **meta** instead of **tenant** and **network**